lib/engine/matchers/part_of_set_matcher.rb

```diff
@@ -0,0 +1,15 @@
+module SplitIoClient
+  class PartOfSetMatcher
+    def self.matcher_type
+      'PART_OF_SET'.freeze
+    end
+
+    def initialize(remote_array)
+      @remote_set = remote_array.to_set
+    end
+
+    def match?(_key, local_array)
+      local_array.to_set.intersect? @remote_set
```

Comment:
if the local_array is what the dev passed along in the code, make sure that:

`local_array intersect @remote` is not `true` when `local_array` is empty.